src/flexmeasures_client/s2/cem.py

```diff
@@ -0,0 +1,268 @@
+            self._logger.warning(f"RM doesn not support `{control_type}` control type.")
+            return
+
+        # RM initialization succeded
+        if self._control_type is not None:
+            message_id = get_unique_id()
+
+            # the callback `update_control_type` will be called upon arrival of a
+            # ReceptionStatus message with status = ReceptionStatusValues.OK
+
+            # register callback in CEM handler
+            if self._control_type in [
+                ControlType.NOT_CONTROLABLE,
+                ControlType.NO_SELECTION,
+            ]:
+                self.register_success_callbacks(
+                    message_id, self.update_control_type, control_type=control_type
+                )
+            else:  # register callback in control mode handler
+                self._control_types_handlers[
+                    self._control_type
+                ].register_success_callbacks(
+                    message_id, self.update_control_type, control_type=control_type
+                )
+
+            await self._sending_queue.put(
+                SelectControlType(message_id=message_id, control_type=control_type)
+            )
+
+    @register(Handshake)
+    def handle_handshake(self, message: Handshake):
+        # TODO: check the version that the RM is using and send a
+        # `selected_protocol_version` that matches the one of the RM
+        # TODO: Return a TBD "CloseConnection" message to close the connection
+
+        return HandshakeResponse(
+            message_id=get_unique_id(), selected_protocol_version=self.__version__
+        )
+
+    @register(ResourceManagerDetails)
+    def handle_resource_manager_details(self, message: ResourceManagerDetails):
+        self._resource_manager_details = message
+
+        if (
+            not self._control_type
+        ):  # initializing. TODO: check if sending resource_manager_details
+            # resets control type
+            self._control_type = ControlType.NO_SELECTION
+
+        return get_reception_status(message)
+
+    @register(PowerMeasurement)
+    def handle_power_measurement(self, message: PowerMeasurement):
+        for power_measurement in message.values:
+            commodity_quantity = power_measurement.commodity_quantity
+
+            if commodity_quantity in self._power_sensors:
+                sensor_id = self._power_sensors[commodity_quantity]
+            else:
+                sensor_id = 1  # TODO: create a new sensor or return ReceptionStatus
+
+            # send measurement
+            self._fm_client.post_measurements(
+                sensor_id,
+                start=message.measurement_timestamp,
+                duration="PT1H",  # TODO: not specified in S2 Protocol
+                values=[power_measurement.value],
+                unit=power_measurement.commodity_quantity,  # TODO: is commodity quantity a unit? for me it's just a the type of POWER # noqa: E501
```

Comment:
Let us collect such mappings in one place. This might be handled here in a way similar to `self._power_sensors`.

S2 `CommodityQuantities` describe a measure and a unit. For example:

```
commodity_quantity = "ELECTRIC.POWER.L1"
measure = "electric power on phase 1"
unit = "W"
```
Note that I derived the above from [this description](https://github.com/flexiblepower/s2-ws-json/blob/main/s2-json-schema/schemas/CommodityQuantity.schema.json).

Only the unit can be communicated through the FlexMeasures API. The concept of measure as a sensor property is no longer modeled in FlexMeasures. We used to have *asset type* for this, but in our data model transition we saw no direct need to move this into a **sensor type** (or what I would now call a **measure**) table.

Reply:
Good idea. 

I think we should collect this info in the [library](https://github.com/flexiblepower/s2-ws-json-python) that we are developing with TNO.

Reply:
Mappings to FlexMeasures terminology have no place there, though, right?